src/sql/platform/query/common/gridDataProvider.ts

```diff
@@ -69,10 +69,12 @@ export async function getResultsString(provider: IGridDataProvider, selection: S
 				let cells = provider.shouldRemoveNewLines()
 					? cellObjects.map(x => removeNewLines(x.displayValue))
 					: cellObjects.map(x => x.displayValue);
-				copyString += cells.join('\t');
-				if (rowIndex < result.resultSubset.rows.length - 1) {
-					copyString += eol;
+
+				let idx = rowIndex + 1;
+				if (copyTable[idx] === undefined) {
```

Comment:
This should be offset based on whether there are headers or not - otherwise the first row will be empty since we'll start at index 1 here

Reply:
Since they're not specifically defined, the first row would simply be elided if its not ever been in use. 

Although this might be an obscure behavior so I'll change it to explicitly use the 0th element if headers aren't picked.